Sources/TuistDependencies/DependenciesController.swift

```diff
@@ -150,3 +157,13 @@ public final class DependenciesController: DependenciesControlling {
         }
     }
 }
+
+extension DependenciesGraph {
+    fileprivate func merging(with other: Self) throws -> Self {
```

Comment:
I'd move this method to [DependenciesGraph.swift file](https://github.com/tuist/tuist/blob/main/Sources/TuistGraph/DependenciesGraph/DependenciesGraph.swift) and make it `public`. 

It could be easier to provide unit tests for the merging logic without DependenciesController context. What is more I can see an opportunity to use this logic in different contexts.